anaBenn/Homework_#2.js

```diff
@@ -0,0 +1,533 @@
+
+function isValid(num,low,high) { 
+    if ((typeof num)!="number") 
+        return NaN;
+    if (num%1 !== 0) 
+        return NaN;
+    if (num<low || num>high) 
+        return NaN;
+    return true;
+}
+
+function rank(card) {
+  return isValid(card,0,51) &&
+  Math.floor(card / 4) + 1;
+}
+
+function suit(card) {
+  var remainder = card % 4;
+  return isValid(card,0,51) && 
+  remainder + 1;
+}
+
+function cardID(rank, suit) {
+  var cardRank = (rank - 1) * 4;
+  var cardSuit = (suit - 1);
+  return isValid(rank,1,13) &&
+  isValid(suit,1,4) &&
+  cardRank + cardSuit;
+}
+
+function color(card) {
+  var theSuit=suit(card); 
+  if (suit(card) === 1 || suit(card) === 2) {
+    return 'red';
+  } else {
+    return 'black';
+  }
+}
+
+function name(card) {
+  var ranksOfCards = ['', 'Ace', 'Two', 'Three', 'Four', 'Five', 'Six', 'Seven', 'Eight', 'Nine', 'Ten', 'Jack', 'Queen', 'King'];
+  var suitsOfCards = ['', 'Hearts', 'Diamonds', 'Spade', 'Clubs'];
+  return ranksOfCards && suitsOfCards && (ranksOfCards[rank(card)] + ' of ' + suitsOfCards[suit(card)]);
+}
+console.log(name(5));
+
+function assert(claim,message) {
+    if (!claim) console.error(message);
+}
+assert(rank(0)===1,  "Test 1 failed");//passed
+assert(rank(3)===1,  "Test 2 failed");//passed
+assert(rank(51)===13,"Test 3 failed");//passed
+assert(suit(0)===1,  "Test 4 failed");//passed
+assert(suit(5)===2,  "Test 5 failed");//passed
+assert(suit(51)===4, "Test 6 failed");//passed
+assert(cardID(1,1)===0,    "Test 7 failed");//passed
+assert(cardID(13,4)===51,  "Test 8 failed");//passed
+assert(cardID(8,3)===30,   "Test 9 failed");//passed
+assert(color(0)==='red',   "Test 10 failed");//passed
+assert(color(2)==='black', "Test 11 failed");//passed
+assert(name(5)==='Two of Diamonds', "Test 12 failed");//passed
+assert(name(51)==='King of Clubs',  "Test 13 failed");//passed
+
+// Hint #3: you can test whether an number n is an integer with (n%1 === 0).
+
+// c) In the file with your solution to part b, extend the existing test suite: write 3 new assertions to test success cases and 
+// 3 more assertions to test failure cases.
+
+//(Please note: Problem 5 is the most important one in this homework, because variations of it will recur in several later 
+//homeworks. The cleaner your code is now, the easier it will be to modify later.)
+
+
+function isValid(num,low,high) { 
+    if ((typeof num)!="number") 
+        return NaN;
+    var remainderNum = (num%1);
+    if (remainderNum !== 0) 
+        return NaN;
+    if (num<low || num>high) 
+        return NaN;
+    return true;
+}
+
+function rank(card) {
+  return isValid(card,0,51) &&
+  Math.floor(card / 4) + 1;
+}
+
+function suit(card) {
+  var remainder = card % 4;
+  return isValid(card,0,51) && 
+  remainder + 1;
+}
+
+function cardID(rank, suit) {
+  var cardRank = (rank - 1) * 4;
+  var cardSuit = (suit - 1);
+  return isValid(rank,1,13) &&
+  isValid(suit,1,4) &&
+  cardRank + cardSuit;
+}
+
+function color(id) {
```

Comment:
Bit of a disconnect here. your (id) isn't moving anywhere in relation to your (card) later on in this function, causing some errors down the line.